src/components/modules/caret.ts

```diff
@@ -548,6 +546,102 @@ export default class Caret extends Module {
     selection.addRange(newRange);
   }
 
+  /**
+   * Detect next position from current position
+   */
+  private detectNext(): Position | false {
+    const { BlockManager, Tools } = this.Editor;
+
+    const nextInput = BlockManager.currentBlock.nextInput;
+    let nextBlock = BlockManager.nextContentfulBlock;
+
+    if (!nextBlock && !nextInput) {
+      /**
+       * If there is a last block and it is default, do nothing
+       * This code allows to exit from the last non-initial tool:
+       * https://github.com/codex-team/editor.js/issues/1103
+       */
+      if (Tools.isDefault(BlockManager.currentBlock.tool)) {
+        return false;
+      }
+
+      /**
+       * If there is no nextBlock, but currentBlock is not default,
+       * insert new default block at the end and navigate to it
+       */
+      nextBlock = BlockManager.insertAtEnd();
+    }
+
+    return this.detectPosition(true, nextBlock, nextInput);
+  }
+
+  /**
+   * Detect previous position from current position
+   */
+  private detectPrevious(): Position | false {
+    const { BlockManager } = this.Editor;
+
+    const previousInput = BlockManager.currentBlock.previousInput;
+    const previousBlock = BlockManager.previousContentfulBlock;
+
+    if (!previousBlock && !previousInput) {
+      return false;
+    }
+
+    return this.detectPosition(false, previousBlock, previousInput);
+  }
+
+  /**
+   * Detect next or previous position from current position
+   *
+   * @param {boolean} isNext - if true, detect next line. Previous otherwise
+   * @param {Block} block - next or previous Block
+   * @param {HTMLElement | undefined} input - next or previous Block
+   */
+  private detectPosition(isNext: boolean, block: Block, input?: HTMLElement): Position {
```

Comment:
`isNext` — I think the `direction` would be a more understandable name. 

```ts
this.detectPosition(Direction.Next, nextBlock, nextInput);
```

Reply:
Fixed.